SensusService/Probes/User/Scripts/ScriptRunner.cs

```diff
@@ -415,52 +446,37 @@ private void StartRandomTriggerCallbacks()
 #error "Unrecognized platform."
 #endif
 
-                // find next future trigger window, ignoring month, day, and year of windows. the windows are already sorted.
+                // find number of days left in participation horizon
```

Comment:
You're right that we'll need to periodically check the callbacks and schedule new ones as the study progresses. The question is how. Previously, the next random script was scheduled after the user completed the current one. You won't be able to pattern after the script tray since it uses threads that will not be allowed to execute when the app is in the background. Keep in mind that we're really only talking about iOS here. On Android everything will happen as desired using the previous implementation since we wake the device up from background, trigger the script, and go back to sleep. The only option I'm seeing for iOS is to check the callbacks and schedule new ones if/when the user completes a script. Like we discussed before, this alleviates the issue but does not solve it. If the user ignores all of the notifications/scripts you schedule for a day, week, etc. (or if the device resets, the app is killed, ...) then no more will be delivered and everything will stop. This might be the best we can do, though we could consider hooking into the significant-change updates as well. We could also think about how push notifications might help here, since we need to add them for other reasons anyway.